### PR TITLE
Update to build and run on jdk9

### PR DESCRIPTION
**Pull request by raboof**
I also added a check to skip the boot classpath when it is not available. This might have to do with http://openjdk.java.net/jeps/261 .

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -1,3 +1 @@
-libraryDependencies += "org.scala-sbt" % "scripted-plugin" % sbtVersion.value
-
-addSbtPlugin("com.thoughtworks.sbt-best-practice" % "sbt-best-practice" % "2.5.0")
```
Comment (Atry) [inline project/plugins.sbt]: The `sbt-best-practice` plugin is necessary for deployment.

**Review by Atry — CHANGES_REQUESTED** [review]
A Pull Request should focus on only one feature. Other changes, like upgrading sbt version for this project itself, should not be included in this PR.

**Comment by raboof** [comment]
@Atry Fair enough - I've reverted the changes related to sbt1 and kept this PR focused on JDK9.

Unfortunately it appears the jdk9 build with sbt 0.13 fails on travis (while it succeeds for me locally), not sure what's going on there.

**Comment by Atry** [comment]
@raboof It seems that we need Scala 2.10.7 for JDK9
https://github.com/scala/scala/pull/6128

**Comment by SethTisue** [comment]
> It seems that we need Scala 2.10.7 for JDK9

and sbt 0.13.17

**Comment by raboof** [comment]
@SethTisue ah - which isn't released yet - good to know!

Do I understand correctly that cross-building for 0.13.16 using the 1.0.2 sbt-launcher does work on jdk9, but building for sbt 0.13 'directly' doesn't until 0.13.17 is out? Or do we need 0.13.17 in both cases?

**Comment by Atry** [comment]
`sbt-api-mappings` should be able to run on JDK9 if we merge this PR and publish a new binary release, despite of the tests on JDK9 is not able to run until sbt 0.13.17 because this plugin itself use 0.13.x.

**Pull request closed without merging**